src/libraries/System.Memory/tests/Span/Overflow.cs

```diff
@@ -78,7 +78,7 @@ public static void SliceStartInt32Overflow()
             {
                 unsafe
                 {
-                    if (!AllocationHelper.TryAllocNative((IntPtr)ThreeGiB, out IntPtr memory))
+                    if (!AllocationHelper.TryAllocNative(unchecked((nint)ThreeGiB), out IntPtr memory))
```

Comment:
Did `IntPtr` used to behave as an unchecked `nint`, hence this update being needed now that `IntPtr` has different behavior?

Reply:
`IntPtr` had no language support while `nint` did. This meant that `(IntPtr)TooLargeConstant` would issue no error while `(nint)TooLargeConstant` would issue a error.

Now that `IntPtr`/`nint` are aliases and have all the same functionality, `IntPtr` issues an error requiring you to acknowledge the potential overflow that existed here and use `unchecked`.